src/robot.coffee

```diff
@@ -247,6 +264,28 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    receiveMiddlewareDone = () ->
+      if cb?
+        process.nextTick -> cb true
+
+    @middleware.receive.execute(
+      {response: new Response(this, message)}
```

Comment:
> The use case is having a receive middleware that blacklists specific users from using any commands. So, it checks for the robot.respond regex, and replies they aren't trusted.

I'm a little confused, do you mean something like this?

```coffeescript
robot.receiveMiddleware (context, next, done) ->
  if context.message.text.match robot.respondPattern /.*/
    # this is a message for the the bot
    if userInBlacklist context.message.user
      # don't let them do anything
    else
      # all good!
      next done
  else
    # not a message for the bot
    next done
```

Assuming yes, I somewhat disagree with that. I feel like that is what listener middleware is for. Middleware shouldn't be doing pattern matching on message text; that is the job of a Listener. That means that you should only be blocking actual commands and not just responding to arbitrary messages using middleware.

Does that make sense? If not, I'll try to explain that again a different way.